Actions/RunPipeline/RunPipeline.ps1

```diff
@@ -129,92 +129,26 @@ try {
 
     if ($baselineWorkflowSHA -and $baselineWorkflowRunId -ne '0' -and $settings.incrementalBuilds.mode -eq 'modifiedApps') {
         # Incremental builds are enabled and we are only building modified apps
-        $headSHA = git rev-parse HEAD
-        Write-Host "Current HEAD is $headSHA"
-        git fetch origin $baselineWorkflowSHA | Out-Host
-        if ($LASTEXITCODE -ne 0) { $host.SetShouldExit(0); throw "Failed to fetch baseline SHA $baselineSHA" }
-        Push-Location $ENV:GITHUB_WORKSPACE
-        Write-Host "git diff --name-only $baselineWorkflowSHA $headSHA"
-        $modifiedFiles = @(git diff --name-only $baselineWorkflowSHA $headSHA | ForEach-Object { "$_".Replace('/', [System.IO.Path]::DirectorySeparatorChar) })
-        if ($LASTEXITCODE -ne 0) { $host.SetShouldExit(0); throw "Failed to diff baseline SHA $baselineSHA with current HEAD $headSHA" }
-        Pop-Location
-        OutputMessageAndArray -message "Modified files" -arrayOfStrings $modifiedFiles
-        $buildAll = Get-BuildAllApps -baseFolder $baseFolder -project $project -modifiedFiles $modifiedFiles
+        try {
+            $modifiedFiles = Get-ModifiedFiles -baselineSHA $baselineWorkflowSHA
+            OutputMessageAndArray -message "Modified files" -arrayOfStrings $modifiedFiles
+            $buildAll = Get-BuildAllApps -baseFolder $baseFolder -project $project -modifiedFiles $modifiedFiles
+        }
+        catch {
+            $buildAll = $true
```

Comment:
Should we log a warning here? Might help debug cases when we unexpectedly end up building all

Reply:
Will add a warning output for this purpose